### DISCUSSION TITLE
filtron.sh installation issues

### DISCUSSION THREAD

**Discussion by pyrenaica** [searx/searx#3412] [Installation issue: Official wiki]
**Version**
_commit_ 117dbd462f39fb4cc2f12e9f7b9acf8d3fff419b
GO version: go1.13.5 linux/amd64

I followed the step-by-step guide and the installation wiki page side by side, Searx installed fine however `filtron.sh install all` doesn't seem to go well as there is no `/usr/local/filtron/go-apps/bin` directory, even though `$GOENV` is set up correctly.

During the installation there seems to be the following issue:
```
# github.com/klauspost/compress/flate
go-apps/src/github.com/klauspost/compress/flate/deflate.go:135:23: cannot convert d.window (type []byte) to type *[32768]byte
go-apps/src/github.com/klauspost/compress/flate/deflate.go:135:56: cannot convert d.window[windowSize:] (type []byte) to type *[32768]byte
go-apps/src/github.com/klauspost/compress/flate/fast_encoder.go:93:28: cannot convert e.hist (type []byte) to type *[32768]byte
go-apps/src/github.com/klauspost/compress/flate/fast_encoder.go:93:63: cannot convert e.hist[offset:] (type []byte) to type *[32768]byte
go-apps/src/github.com/klauspost/compress/flate/huffman_bit_writer.go:794:17: cannot convert w.literalFreq[:] (type []uint16) to type *[256]uint16
go-apps/src/github.com/klauspost/compress/flate/huffman_bit_writer.go:796:16: cannot convert w.literalFreq[256:] (type []uint16) to type *[32]uint16
```

**How To Reproduce**
Simply run `./filtron.sh install all`

**Expected behavior**
Installation of Filtron and the activation of the Systemd unit.

**Additional context**
Installing it in a Debian11 LXC container, used pip and virtualenv.

**Comment by kvch** [comment]
Hi, your go version seems to old. Filtron requires Golang 1.19. Please update your Go version.

**Comment by minami-o** [comment]
Looks like the filtron.sh script is incorrect indeed: it downloads v1.13.5 of go and uses the `go get` command which is deprecated (at least with 1.19, you should use `go install` instead).
You can manually update the script and it will solve the problem.
